.github/actions/checksum-check/action.yml

```diff
@@ -0,0 +1,47 @@
+name: Checksum Check
+inputs:
+  checksum-name:
+    type: string
+    required: true
+    description: The name of the GitHub artifact containing the checksum that will be compared against
+  checksum-location:
+    type: string
+    required: true
+    description: Location of the GitHub artifact containing the checksum that will be compared against
+  checked-against:
+    type: string
+    required: false
+    description: Explicit config tag to checkout the 'ground truth' checksum. Defaults to last major truth update.
+outputs:
+  result:
+    value: ${{ steps.check.outputs.result }}
+    description: The result of the comparison between the artifact and the latest ground truth
+  ground-truth-version:
+    value: ${{ steps.ground-truth.outputs.version }}
+    description: The ground truth version compared against the given artifact
+runs:
+    - name: Determine Tag to Check Against
+      shell: bash
+      id: ground-truth
+      run: |
+        if [[ "${{ inputs.checked-against }}" == "" ]]; then
+          echo "version=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT
+        else
+          echo "version=${{ inputs.checked-against }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Compare Checksum against ${{ steps.ground-truth.outputs.version }}
+      # we use the full `bash {0}` syntax instead of `bash` because we don't want the
+      # assumed `-e` (exit on error): the nonzero exit code on `diff` is actually used later on.
+      shell: bash {0}
+      id: check
+      run: |
+        git checkout ${{ steps.ground-truth.outputs.version }}
+        diff ${{ inputs.checksum-location }}/CHECKSUM CHECKSUM
```

Comment:
I don't think the name of the checksum file should be hard-coded. I think we should let the called specify what, and where, it is. We could have something like `inputs.new-checksum-path` and `inputs.truth-checksum-path` and specify the location and filename, or two locations and a common filename, but that requires three arguments, so probably the former is better. Don't mind what you call them.